### DISCUSSION TITLE
How long vercel caches old deployments ?

### DISCUSSION THREAD

**Discussion by besoeasy** [vercel/vercel#4982] [Help] [answered]
hey, i recently started using vercel and i see a deployment for each of my commits. How long are deployments cached ? stored ? 

I don't want to hog vercel resources are they deleted automatically or do we have to purge them manually ?

**Comment by styfle** [comment]
You don't need to worry about old deployments. They are archived after they become inactive for some time. They are also not indexed by search engines so its safe to ignore these.

**Reply by tonyxiao** [reply]
@styfle What about old deployment that have features that should not have been released? Do we need to actively go delete all of them or will they be automatically deleted / taken down after some time? If so how long is that time?

**Reply by styfle** [reply]
If you want to permanently delete the deployment, it must be manually deleted. There is no automatic deletion at this time.

**Reply by tonyxiao** [reply]
Environment variables are preserved per deployment right? If we want to just permannetly break old deployments we can for example change database pw and not worry about manually going back to delete each of them?

**Reply by styfle** [reply]
Correct

> Changes to Environment Variables are not applied to previous deployments, they only apply to new deployments.

https://vercel.com/docs/concepts/projects/environment-variables

**Reply by besoeasy** [reply]
I believe it would be really helpful for users if you could allow setting a time-to-live (TTL) for old deployments, so that they can be automatically deleted after a certain period of time. This would help avoid cluttering up the deployment list and make it easier for users to manage their deployments.

Thank you for considering this suggestion. Keep up the great work!

**Comment by leerob** [comment, marked as answer]
You can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).
